ls1intum/Artemis
# [Enhancement] Programming Assessment refactoring routes to use submissionId

### Checklist
- [x] Server: I extended and adapted multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Client: I adapted multiple integration tests (Jest) related to the features (with a high test coverage)

### Motivation and Context


All exercises types except of programming exercises use the submissionId for routing and fetching data. Programming exercises use the participationId as in the past a submission could not have multiple results. This change makes it possible to switch now from the participationId to use the submissionId like for the other exercise types.

### Description

- Change participationId to submissionId in routes and some endpoints
- Refactor CodeEditorTutorContainer
- Remove not needed endpoints and resolvers

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create several programming exercises with manual feedback (exam, team, normal) and also with and without SCA and SGI. Please try different combinations.
4. Participate in the exercise and assess it afterwards, everything should work as always.
4.1 Access the assessment from assessment dashboard, exercise scores and exercise submission view
5. Check that the routes now use submissionId instead of participationId.
6. Check that from the students perspective everything works fine.
7. Complaint as student and evaluate the complaint, everything should work as always

### Test Coverage



Nothing changed

### Screenshots



New URL:
<img width="1671" alt="image" src="https://user-images.githubusercontent.com/41108487/113810864-69f6aa00-976b-11eb-96c6-b791370ae1ec.png">

After reentering the assessment all feedbacks are correctly loaded:
<img width="1677" alt="image" src="https://user-images.githubusercontent.com/41108487/113811205-1173dc80-976c-11eb-9814-3c53264fa2a3.png">

Student View:
<img width="1654" alt="image" src="https://user-images.githubusercontent.com/41108487/113811275-2e101480-976c-11eb-9de2-c3eeed720e70.png">
